app/models/billboard.rb

```diff
@@ -101,6 +103,28 @@ def self.for_display(area:, user_signed_in:, user_id: nil, article: nil, user_ta
     end
   end
 
+  def self.weighted_random_selection(relation)
+    base_query = relation.to_sql
+    random_val = connection.select_value("SELECT RANDOM()").to_f
```

Comment:
Is there a reason this _needs_ to come from the DB? It seems like we could probably generate a random value in Ruby and save a query here.